lib/wlib/strings/DynamicString.h

```diff
@@ -0,0 +1,476 @@
+#ifndef WLIB_DYNAMICSTRING_H
+#define WLIB_DYNAMICSTRING_H
+
+#include "stdint.h"
+#include "string.h"
+#include <math.h>
+#include "memory/Memory.h"
+
+class DynamicString {
+private:
+	char * m_buffer;
+public:
+	/**
+	 * Default constructor creates string with no characters
+	 */
+	DynamicString(){
+		m_buffer = (char *)memory_alloc(sizeof(char));
+		m_buffer[0] = '\0';
+	}
+
+	/**
+	 * Constructor creates string using character array
+	 *
+	 * @param str char string
+	 */
+	DynamicString(const char* str){
+		uint16_t size = (uint16_t) strlen(str);
+		m_buffer = (char *)memory_alloc((size + 1) * sizeof(char));
+		strncpy(m_buffer, str, size);
+		m_buffer[size] = '\0';
+	}
+
+	/**
+	 * Constructor creates string using DynamicString object
+	 *
+	 * @param str @code DynamicString object
+	 */
+	DynamicString(const DynamicString &str){
+		DynamicString(str.c_str());
+	}
+
+	/**
+	 * Assign operator assigns current object to given object
+	 *
+	 * @param str @code DynamicString object
+	 * @return current object
+	 */
+	DynamicString operator= (const DynamicString &str){
+		return operator=(str.c_str());
+	}
+
+	/**
+	 * Assign operator assigns current object to given character string
+	 * @param str
+	 * @return current object
+	 */
+	DynamicString operator= (const char* str){
+		uint16_t size = (uint16_t) strlen(str);
+		m_buffer = (char *)memory_realloc(m_buffer, (size + 1) * sizeof(char));
+		strncpy(m_buffer, str, size);
+		m_buffer[size] = '\0';
+
+		return *this;
+	}
+
+	/**
+	 * Provides current length of string
+	 *
+	 * @return string length
+	 */
+	uint16_t length() const{
+		uint16_t i = 0;
+		while (m_buffer[i] != 0) i++;
+		return i;
+	}
+
+	/**
+	 * Clears the string such that there are no characters left in it
+	 */
+	void clear(){
+		m_buffer[0] = '\0';
+	}
+
+	/**
+	 * Element access operator gives access to character at @code pos
+	 *
+	 * @param pos the position of the character
+	 * @return character at @code position
+	 */
+	char &operator[](uint16_t pos){

```

Comment:
`uint16_t` should be replaced with `size_type` in `Types.h` everywhere in this file

Reply:
no use uint16_t everywhere except where you care about size. The length, capacity, for loops you get the point. This is basically `size_t`

Reply:
True but that would mostly apply here wouldn't it? Position in an array with length `size_type` should also be of type `size_type`

Reply:
So replace all uint16_t with size_type?

Reply:
Ye